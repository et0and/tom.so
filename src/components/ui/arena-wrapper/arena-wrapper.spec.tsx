import { render } from "@testing-library/react";
import ArenaWrapper from "./arena-wrapper";

describe("<ArenaWrapper />", () => {
  it("renders correctly with channel slug", () => {
    const channelSlug = "example-channel";
    const { container } = render(<ArenaWrapper channelSlug={channelSlug} />);

    expect(container).toMatchSnapshot();
  });

  it("creates iframe with correct src based on channelSlug", () => {
    const channelSlug = "example-channel";
    const { getByTitle } = render(<ArenaWrapper channelSlug={channelSlug} />);

    const iframe = getByTitle(`Are.na channel ${channelSlug}`);
    expect(iframe.getAttribute("src")).toBe(
      `https://www.are.na/tom/${channelSlug}/embed`,
    );
    expect(iframe.getAttribute("width")).toBe("100%");
    expect(iframe.getAttribute("height")).toBe("590");
  });
});
