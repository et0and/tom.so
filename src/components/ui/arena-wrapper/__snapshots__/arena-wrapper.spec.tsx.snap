// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArenaWrapper /> renders correctly with channel slug 1`] = `
<div>
  <iframe
    class="bg-slate-300"
    height="590"
    src="https://www.are.na/tom/example-channel/embed"
    style="border: 1px solid #e5e5e5;"
    title="Are.na channel example-channel"
    width="100%"
  />
</div>
`;
